projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -84,55 +97,195 @@ export class IgxTimePickerComponent implements ControlValueAccessor, OnInit, OnD
         }
     }
 
+    /**
+     * An accessor that returns the value of `igx-time-picker` component.
+     * ```html
+     *@ViewChild("MyPick")
+     *public pick: IgxTimePickerComponent;
+     *ngAfterViewInit(){
+     *    let pickSelect = this.pick.value;
+     * }
+     * ```
+     */
     get value(): Date {
         return this._value;
     }
 
+    /**
+     * An @Input property that allows you to disable the `igx-time-picker` component. By default `isDisabled` is set to false.
+     * ```html
+     * <igx-time-picker [isDisabled]="true" [vertical]="true" format="h:mm tt" ></igx-time-picker>
+     * ```
+     */
     @Input()
     public disabled = false;
 
+    /**
+     * An @Input property that renders OK button with custom text. By default `okButtonLabel` is set to OK.
+     * ```html
+     * <igx-time-picker okButtonLabel='SET' [value]="date" format="h:mm tt"></igx-time-picker>
+     * ```
+     */
     @Input()
     public okButtonLabel = 'OK';
 
+    /**
+     * An @Input property that renders cancel button with custom text.
+     * By default `cancelButtonLabel` is set to Cancel.
+     * ```html
+     * <igx-time-picker cancelButtonLabel='Exit' [value]="date" format="h:mm tt"></igx-time-picker>
+     * ```
+     */
     @Input()
     public cancelButtonLabel = 'Cancel';
 
+    /**
+     * An @Input property that gets/sets the delta by which hour and minute items would be displayed.
```

Comment:
@alexalex972 "... by which hour and minute items would be changed when the user presses the Up/Down keys"
Note: "Not sure whether it's the keys. Please check it when this option takes place."